addons/html_editor/static/src/editor/core/delete_plugin.js

```diff
@@ -441,907 +556,168 @@ export class DeletePlugin extends Plugin {
-        // 12 and 34 text node.
-        if (offsetLimit === undefined) {
-            while (node && !isBlock(node)) {
-                node = node.nextSibling;
-                currentNodeIndex++;
-            }
-        } else {
-            currentNodeIndex = offsetLimit;
-        }
-        const [cursorNode, cursorOffset] = moveNodes(
-            ...moveDest,
-            targetNode,
-            targetOffset,
-            currentNodeIndex
-        );
-        return [moveDest, alreadyMoved, cursorNode, cursorOffset];
-    }
-    deleteElementBackwardPropagate(nextSibling, moveDest, cursorNode, cursorOffset, alreadyMoved) {
-        if (
-            cursorNode.nodeType === Node.TEXT_NODE &&
-            (cursorOffset === 0 || cursorOffset === cursorNode.length)
-        ) {
-            cursorOffset = childNodeIndex(cursorNode) + (cursorOffset === 0 ? 0 : 1);
-            cursorNode = cursorNode.parentNode;
-        }
-        if (cursorNode.nodeType !== Node.TEXT_NODE) {
-            const { cType } = getState(cursorNode, cursorOffset, DIRECTIONS.LEFT);
-            if ((cType & CTGROUPS.BLOCK && !alreadyMoved) || cType === CTYPES.BLOCK_OUTSIDE) {
-                this.deleteElementBackward({
-                    targetNode: cursorNode,
-                    targetOffset: cursorOffset,
-                    alreadyMoved,
-                });
-            } else if (!alreadyMoved) {
-                // When removing a block node adjacent to an inline node, we need to
-                // ensure the block node induced line break are kept with a <br>.
-                // ex : <div>a<span>b</span><p>[]c</p>d</div> => deleteBakward =>
-                // <div>a<span>b</span>[]c<br>d</div> In this case we cannot simply
-                // merge the <p> content into the div parent, or we would lose the
-                // line break located after the <p>.
-                const cursorNodeNode = cursorNode.childNodes[cursorOffset];
-                const cursorNodeRightNode = cursorNodeNode ? cursorNodeNode.nextSibling : undefined;
-                if (
-                    cursorNodeRightNode &&
-                    cursorNodeRightNode.nodeType === Node.TEXT_NODE &&
-                    nextSibling === cursorNodeRightNode
-                ) {
-                    moveDest[0].insertBefore(document.createElement("br"), cursorNodeRightNode);
+        return false;
+    }
+
+    // Returns the previous visible position (ex: a previous character, the end
+    // of the previous block, etc.).
+    findPreviousPosition(node, offset, blockSwitch = false) {
+        // Look for a visible character in text node.
+        if (node.nodeType === Node.TEXT_NODE) {
+            // @todo @phoenix: write tests for chars with size > 1 (emoji, etc.)
+            // Use the string iterator to handle surrogate pairs.
+            let index = offset;
+            const chars = node.textContent.slice(0, index);
+            for (const char of [...chars].reverse()) {
```

Comment:
not like this...

Reply:
I got rid of the call to `reverse`... but I still iterate twice over the string (because the string iterator handles chars with len > 1, and there's no reverse iterator for string...)

Reply:
(3 times actually: slice, string to array, and popping in the while loop...)